setup.php

```diff
@@ -356,10 +356,11 @@ function cloneClicked() {
                 echo " <td class='text'>(Clone the source site's database instead of creating a fresh one.)</td>\n";
                 echo "</tr>\n";
             }
+            $randomusername = chr(rand(65, 90)) . chr(rand(65, 90)) . chr(rand(65, 90)) . "-admin-" . rand(0, 9) . rand(0, 9);
 
             echo "<TR VALIGN='TOP' class='noclone'><TD COLSPAN=2><font color='red'>OPENEMR USER:</font></TD></TR>";
-            echo "<TR VALIGN='TOP' class='noclone'><TD><span class='text'>Initial User:</span></TD><TD><INPUT SIZE='30' TYPE='TEXT' NAME='iuser' VALUE='admin'></TD><TD><span class='text'>(This is the login name of user that will be created for you. Limit this to one word.)</span></TD></TR>
-<TR VALIGN='TOP' class='noclone'><TD><span class='text'>Initial User Password:</span></TD><TD><INPUT SIZE='30' TYPE='PASSWORD' NAME='iuserpass' VALUE=''></TD><TD><span class='text'>(This is the password for the initial user account above.)</span></TD></TR>
+            echo "<TR VALIGN='TOP' class='noclone'><TD><span class='text'>Initial User:</span></TD><TD><INPUT SIZE='30' TYPE='TEXT' NAME='iuser' VALUE='".$randomusername."' minlength=12></TD><TD><span class='text'>(This is the login name of user that will be created for you. Limit this to one word with at least 12 characters.)</span></TD></TR>
+<TR VALIGN='TOP' class='noclone'><TD><span class='text'>Initial User Password:</span></TD><TD><INPUT SIZE='30' TYPE='PASSWORD' NAME='iuserpass' VALUE='' minlength=12></TD><TD><span class='text'>(This is the password for the initial user account above - it should be at least 12 characters long.)</span></TD></TR>
```

Comment:
for both lines above, put singlequotes around the 12 in minlength attribute

Reply:
Lets instruct that the Initial User Password should be composed of letters and numbers

Reply:
But, as of now there is no validation on password for that.